scripts/zkai_d64_block_receipt_composition_gate.py

```
@@ -0,0 +1,958 @@
+            raise D64BlockReceiptError("duplicate slice in slice chain")
+        actual_ids.append(slice_id)
+        by_slice[slice_id] = item
+    if actual_ids != expected_ids:
+        raise D64BlockReceiptError("slice chain order or membership mismatch")
+    for spec in SLICE_SPECS:
+        item = by_slice[spec.slice_id]
+        expect_equal(item.get("schema"), spec.schema, f"{spec.slice_id} schema")
+        expect_equal(item.get("decision"), spec.decision, f"{spec.slice_id} decision")
+        expect_equal(item.get("proof_backend_version"), spec.proof_version, f"{spec.slice_id} proof version")
+        if not isinstance(item.get("source_commitments"), dict):
+            raise D64BlockReceiptError(f"{spec.slice_id} source commitments must be an object")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)